doc/modules/multiclass.rst

```diff
@@ -14,52 +14,116 @@ Multiclass and multilabel algorithms
-  apples, or pears. Multiclass classification makes the assumption that each
-  sample is assigned to one and only one label: a fruit can be either an
-  apple or a pear but not both at the same time.
-
-- **Multilabel classification** assigns to each sample a set of target
-  labels. This can be thought as predicting properties of a data-point
-  that are not mutually exclusive, such as topics that are relevant for a
-  document. A text might be about any of religion, politics, finance or
-  education at the same time or none of these.
-
-- **Multioutput regression** assigns each sample a set of target
-  values.  This can be thought of as predicting several properties
-  for each data-point, such as wind direction and magnitude at a
-  certain location.
-
-- **Multioutput-multiclass classification** and **multi-task classification**
-  means that a single estimator has to handle several joint classification
-  tasks. This is both a generalization of the multi-label classification
-  task, which only considers binary classification, as well as a
-  generalization of the multi-class classification task.  *The output format
-  is a 2d numpy array or sparse matrix.*
-
-  The set of labels can be different for each output variable.
-  For instance, a sample could be assigned "pear" for an output variable that
-  takes possible values in a finite set of species such as "pear", "apple"; 
-  and "blue" or "green" for a second output variable that takes possible values
-  in a finite set of colors such as "green", "red", "blue", "yellow"...
-
-  This means that any classifiers handling multi-output
-  multiclass or multi-task classification tasks,
-  support the multi-label classification task as a special case.
-  Multi-task classification is similar to the multi-output
-  classification task with different model formulations. For
-  more information, see the relevant estimator documentation.
+- **Multiclass classification**: classification task which labels each
+  sample as **one** of a number of classes, where the number of classes is
+  greater than 2. Each sample can only be labelled as one class.
+
+  For example, classification using features extracted from a set of images of
+  fruit, where each image may either be of an orange, an apples, or a pear.
+  Each image is one sample and is labelled as one of the 3 possible classes.
+  Multiclass classification makes the assumption that each sample is assigned
+  to one and only one label - one sample cannot, for example, be both a pear
+  and an apple at the same time.
+  
+  - Valid :term:`multiclass` representations for
+    :func:`~utils.multiclass.type_of_target` (`y`) are:
+
+    - 1d or column vector containing more than two discrete values.
+    - sparse :term:`binary` matrix of shape ``(n_samples, n_classes)`` with a
+      single element per row, where each column represents one class.
+
+
+- **Multilabel classification**: classification task which labels each sample
```

Comment:
It would be worth explaining why/how this differs from just running n binary classification tasks?

Also the new example is good but we could also keep the old one about documents and topics, since it's more a real-world example.